### PR TITLE
Let's discuss this pull request

### PR DESCRIPTION
**Pull request by MatthijsKamstra**
I started an update, but the same time you updated.. So that might make this the reason we can't merge automatically.

Let me know the parts you want to have changed for this merge.

The example folder is the main reason for the pull request. If you okay it, I wil change the urls to yours

Great work btw!

grt
Matthijs

### CODE REVIEW

**Comment by markknol** [comment]
In the same hour I also got this request: https://github.com/markknol/hx-quicksettings/pull/3
Can you make a new PR which doesn't conflict? I'm fine with everything you did :+1:

**Comment by MatthijsKamstra** [comment]
Will look into!

File: quicksettings/QuickSettings.hx [hunk 1]
```diff
@@ -1,189 +1,343 @@
...
    * creates a button
    */
   public function addButton(title:String, callback:Void->Void):Void;
-  
+
   /**
    * creates informational text:String
+   * deprecated. Identical to addHTML
    */
+  @:deprecated("addInfo is deprecated, use addHTML instead")
```
Comment (markknol) [inline quicksettings/QuickSettings.hx:109]: nice

File: quicksettings/QuickSettings.hx [hunk 2]
```diff
@@ -1,189 +1,343 @@
...
+  @optional var snapToGrid : Bool;            // optional bool,   default false
+  @optional var gridSize : Int;               // optional number  default 0
+  @optional var controls : Array<QControls>;   // optional array of control objects
+};
+
+typedef QControls =
+{
+  var type : String;                          // required string
+  var title : String;                         // required string
+  @optional var value : Dynamic;              // optional value:
```
Comment (markknol) [inline quicksettings/QuickSettings.hx]: I think it is better to use a block comment `/** comment **/` before each var for nicer code hints / api docs.

File: quicksettings/QuickSettings.hx [hunk 3]
```diff
@@ -1,189 +1,343 @@
...
+}
+
+typedef OnFileChosen = {
+  var type : String;
+  var size : String;
+  var name : String;
+  var lastModifiedDate : String;
 }
+
+// not sure how usefull...
```
Comment (markknol) [inline quicksettings/QuickSettings.hx]: I think its good

File: README.md [hunk 4]
```diff
@@ -1,25 +1,62 @@
...
+
+
+```html
+<!-- for example: use the quicksettings_tiny_black.min.css -->
+<link rel="stylesheet" type="text/css" href="quicksettings_tiny_black.min.css">
 <script type="text/javascript" src="quicksettings.js"></script>
 ```
 
+
+# haxelib local use
```
Comment (markknol) [inline README.md:28]: I didn't even know this was possible. :+1:

File: quicksettings/QuickSettings.hx (hunk 2 above)
Comment (MatthijsKamstra) [reply quicksettings/QuickSettings.hx]: does that work with typedef? 
and do you mean this?

```
typedef QSettings =
{
      /** optional string, default "QuickSettings" **/
      @optional var title : String;               
...
```

File: README.md (hunk 4 above)
Comment (MatthijsKamstra) [reply README.md:28]: yes, too many hidden features... 

Andyli showed me `haxe install build.hxml` recently, so nice!

File: quicksettings/QuickSettings.hx (hunk 2 above)
Comment (markknol) [reply quicksettings/QuickSettings.hx]: Yep exactly that

**Comment by MatthijsKamstra** [comment]
Okay, lets try this again

**Comment by markknol** [comment]
It still has conflicts :open_mouth: I think you need to merge the master and resolve the conflicting issues.

**Comment by markknol** [comment]
I resolved some conflicts and merged, can you double check too?

**Comment by MatthijsKamstra** [comment]
looks okay to me: thx

**Comment by markknol** [comment]
Ok great. I also Haxified some parts of the demo to make it more pretty.

**Comment by MatthijsKamstra** [comment]
the first time you open the preview page it doesn't show the quicksettings
we need 

```
        document.addEventListener("DOMContentLoaded", function(event) {
            init();
        });
```

not sure what "Haxified" means, but if it means that I am a sloppy coder: you are right 😛

**Comment by MatthijsKamstra** [comment]
oh now I see `${file.name}`, yes its a literal copy of the original js code

**Pull request merged**